truffle/src/main/java/org/jruby/truffle/core/rope/SubstringRope.java

```diff
@@ -62,7 +62,13 @@ public Rope withEncoding(Encoding newEncoding, CodeRange newCodeRange) {
 
     @Override
     public byte getByteSlow(int index) {
-        return child.getByteSlow(index + offset);
+        int idx = index + offset;
+        Rope nextChild = child;
+        while (nextChild instanceof SubstringRope) {
```

Comment:
While there's nothing wrong with this check, when we actually create `SubstringRope` instances we always collapse substrings of substrings (see https://github.com/jruby/jruby/blob/5c97ec03aefe6671cb7a2ad61b88a50bf3319b33/truffle/src/main/java/org/jruby/truffle/core/rope/RopeNodes.java#L119-L124 for details).

And since you only handle that one case here, I think whatever you were looking to accomplish devolves into the code you replaced.